src/sio_tactician.c

```diff
@@ -0,0 +1,324 @@
+        TacticianDrawCharacters(proc);
+        proc->child1->unk40 = proc->cur_len * 7;
+    }
+    BG_EnableSyncByMask(BG0_SYNC_BIT | BG1_SYNC_BIT | BG2_SYNC_BIT | BG3_SYNC_BIT);
+}
+
+void SioUpdateTeam(char * str, int team)
+{
+    int i;
+    struct Unit * buffer = GetUnit(FACTION_RED + 1);
+    for (i = 0; i < 5; i++)
+        ClearUnit(buffer + i);
+
+    for (i = 0; i < 5; i++)
+    {
+        u8 pid = gSioPidPool.pids[i];
+        if (pid != 0)
+        {
+            struct Unit * unit = GetUnitFromCharId(pid);
+            if (!(unit->state & US_NOT_DEPLOYED))
+            {
+                SetUnitStatus(unit, UNIT_STATUS_NONE);
+                unit->state = 0;
+                MemCpy(unit, buffer + i, sizeof(struct Unit));
+            }
+        }
+    }
+    WriteMultiArenaSaveTeam(team, buffer, str);
+}
+
+void sub_80449E8(struct ProcTactician * proc, int idx, const struct TacticianTextConf * conf)
+{
+    int str_idx;
+    u16 ret;
+    const struct TacticianTextConf * conf2;
+
+    ret = conf->unk36[idx];
+    conf2 = gTacticianTextConf + conf->unk36[idx];
+
+    str_idx = proc->line_idx * 3;
+
+    while (*conf2->str[str_idx] == '\0')
+    {
+        ret = conf2->unk36[idx];
+        conf2 = gTacticianTextConf + conf2->unk36[idx];
+    }
+    proc->conf_idx = ret;
+}
+
+void TacticianTryAppendChar(struct ProcTactician * proc, const struct TacticianTextConf * conf)
+{
+    int cur_len;
+
+    if (proc->cur_len < proc->max_len)
+    {
+        SioPlaySoundEffect(2);
+        SioStrCpy(conf->str[proc->line_idx * 3], &proc->str[proc->cur_len]);
+
+        proc->unk4C[proc->cur_len] = (0x3FFF & proc->conf_idx) | ((3 & proc->line_idx) << 14);
+        cur_len = proc->cur_len + 1;
+
+        if (cur_len < proc->max_len)
+            proc->cur_len = cur_len;
+        else
+            proc->conf_idx = 5;
+
+        TacticianDrawCharacters(proc);
+        proc->unk39 = 0;
+    }
+    else
+    {
+        SioPlaySoundEffect(0);
+    }
+}
+
+void TacticianTryDeleteChar(struct ProcTactician * proc, const struct TacticianTextConf * conf)
+{
+    int cur_len;
+
+    if (proc->cur_len != 0)
+    {
+        SioPlaySoundEffect(2);
+
+        if (proc->unk4C[proc->cur_len] == 0)
+            proc->cur_len--;
+
+        *(proc->str + proc->cur_len) = 0;
+        proc->unk4C[proc->cur_len] = 0;
+        proc->unk39 = 0;
+
+        TacticianDrawCharacters(proc);
+    }
+    else
+    {
+        SioPlaySoundEffect(0);
+    }
+}
+
+void SaveTactician(struct ProcTactician * proc, const struct TacticianTextConf * conf)
+{
+    if (proc->str[0] != '\0')
+    {
+        SioPlaySoundEffect(2);
+
+        if (CheckInLinkArena())
+            SioUpdateTeam(proc->str, gLinkArenaSt.unk_03);
+        else
+            SetTacticianName(proc->str);
+
+        Proc_Break(proc);
+    }
+    else
+    {
+        SioPlaySoundEffect(0);
+    }
+}
+
+#if 0
+bool sub_8044B78(struct ProcTactician * proc, const struct TacticianTextConf * conf, u32 c, int d)
```

Comment:
Is there a decomp.me scratch for this one?

Reply:
Oh I forgot to put on it, now matched by @jiangzhengwenjz